### DISCUSSION TITLE
Windows Terminal and PowerToys Run

### DISCUSSION THREAD

**Discussion by eggbean** [chrisant996/clink#476] [General]
I have just realised that the [PowerToys Run](https://learn.microsoft.com/en-us/windows/powertoys/run) feature allows you to open a Quake style pulldown terminal window, which is pretty cool.

The only problem is that it doesn't start clink when cmd.exe is started, which is terrible. Does anybody know how to make it run clink?

**Comment by eggbean** [comment]
I don't know why it wasn't working before, but it's fine now.

**Comment by chrisant996** [comment]
It depends on whether Clink is configured for autorun (if Clink was installed using the setup program with the "Autorun when cmd.exe starts" box checked, or if the `clink autorun install` command has been used).

- If configured for autorun, Clink is injected automatically.
- If not configured for autorun, Clink must be injected explicitly by running `clink inject`.  Since Quake mode starts the default tab profile in Windows Terminal, adding `& clink inject` to the end of the command line in the tab profile can inject Clink.

If you encounter a case where you believe an attempt was made to inject Clink but it seems to have failed, then reviewing the `clink.log` file might have some clues as to why.  Running `clink info` reports the location of the `clink.log` file.  (Note that each time a new instance of Clink starts, it clears the log file -- so if you want to check the log file, avoid starting another Clink to do so, or it will clear the log file.)

**Reply by eggbean** [reply]
Thanks.
